### PR TITLE
vnext update gulp

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Temporary put dirty handling of "tag" into gulp.json.

It creates a temporary md file which contains only the input files from the specified "tag".

Usage
`gulp codegen --spec-root "C:\github\azure-rest-api-specs" --autorest-java "C:\github_fork\autorest.java\fluentgen" --projects resources`

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ var autoRestExe;
 gulp.task('codegen', function(cb) {
     if (autoRestVersion.match(/[0-9]+\.[0-9]+\.[0-9]+.*/) ||
         autoRestVersion == 'latest') {
-            autoRestExe = 'autorest ---version=' + autoRestVersion;
+            autoRestExe = 'autorest-beta ---version=' + autoRestVersion;
```
Comment (yaohaizh) [inline gulpfile.js]: Need update package.json

File: gulpfile.js [hunk 2]
```diff
@@ -107,8 +107,13 @@ var codegen = function(project, cb) {
         deleteFolderRecursive(sourcesToDelete);
     }
 
-    console.log('Generating "' + project + '" from spec file ' + specRoot + '/' + mappings[project].source);
-    var generator = '--fluent';
+    let readmeFile = specRoot + '/' + mappings[project].source;
```
Comment (yaohaizh) [inline gulpfile.js]: use path.join for multi platform

File: gulpfile.js (hunk 1 above)
Comment (weidongxu-microsoft) [reply gulpfile.js]: I think package.json is not needed. It just calls autorest-beta on exec. No dependency on nodejs.

File: gulpfile.js (hunk 2 above)
Comment (weidongxu-microsoft) [reply gulpfile.js]: There were a comment nearby: "path.join won't work if specRoot is a URL", so it uses '/'.

For now the dirty approach would need the spec on local filesystem to work. But still keep it this way. Some day we might no longer need this hacking and re-enable the URL.

File: gulpfile.js [hunk 3]
```diff
@@ -144,6 +150,42 @@ var codegen = function(project, cb) {
     return execa(cmd, [], { shell: true, stdio: "inherit" });
 };
 
+var findTag = function(stringContainsTag) {
+    let regex = new RegExp('--tag=(\\S+)');
```
Comment (yaohaizh) [inline gulpfile.js]: const

**Pull request merged**